internal/api/kubernetes/client.go

```diff
@@ -578,15 +578,15 @@ func GetRestConfig(ctx context.Context, path string) (*rest.Config, error) {
 	// is empty, but will issue a warning that we can suppress by checking for
 	// that condition ourselves and calling rest.InClusterConfig() directly.
 	if path == "" {
-		logger.Debug("loading in-cluster REST config")
+		logger.V(1).Info("loading in-cluster REST config")
```

Comment:
I feel uncertain about anywhere where we see something like `logger.V(1)`. Doesn't the call to `V()` _change_ the verbosity level? Except for in the startup logs, (as mentioned [here](https://github.com/akuity/kargo/pull/1791/files#r1556679011)) we want to use the level specified by whoever installed Kargo. i.e. If obtained from `logging.LoggerFromContext`, it should already be configured to the correct verbosity.

Reply:
🤦‍♂️ You're right. Although I'm comfortable with the notion of an abstraction over the underlying logger (mainly from my Java days), having not used logr before, I wasn't at all prepared for how wildly different the interface is from what I am accustomed to. My initial reaction is that needing to remember numeric levels as I code seems painful in comparison to `.Debug()` or `.Warn()`. Let me cozy up with the logr docs and try to wrap my head around why this is supposed to be better before commenting further.

Reply:
@hiddeco pointed out that realistically, we'd use constants we defined ourselves for the levels.

I'm still left feeling that `logger.V(logging.DEBUG).Info(...)` is all at once wordier _and_ less expressive than `logger.Debug(...)`.

_I'm totally bought into the benefits of logr, but just cannot get over the interface._

Wondering what @jessesuen and @hiddeco might think about a thin wrapper around `logr.Logger` in our existing `logging` package. `logging.DefaultLogger()` and `logging.LoggerFromContext(ctx)` could be modified to return something that looks like this:

```golang
type Logger struct {
	*logr.Logger
}

func (l *Logger) Debug(msg string, keysAndValues ...any) {
	l.Logger.V(1).Info(msg, keysAndValues...)
}

// etc...

func (l *Logger) WithValues(keysAndValues ...any) *Logger {
	newLogger := l.Logger.WithValues(keysAndValues...)
	return &Logger{Logger: &newLogger}
}
```

To be clear... I'm _not_ proposing an "abstraction over an abstraction." I'm proposing fully taking advantage of the logr abstraction (i.e. we can still swap logging back ends no problem), but merely enhancing their interface _without obscuring access_ in case we sometimes need to "drop-down" to it.

Reply:
On one hand, I feel averse of this idea because it does not allow you to [log errors](https://github.com/go-logr/logr/blob/master/logr.go#L294-L302) at debug level, which a real `logr.Logger` implementation does allow you to do.

I also think (and know based on experience), the problem can largely be worked around by using a modular approach in your application architecture. Allowing you to e.g. inject a `logr.Logger` instance at a predefined level, so that you have to think less about the level which you are logging at.

On the other hand, is the question how often you really have the desire to log an error at debug level, and [isn't this approach really uncommon in our landscape](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/b918c3636f5dcc030bbbc5191e757e37df87c682/pkg/logger/logger.go).

Reply:
> it does not allow you to [log errors](https://github.com/go-logr/logr/blob/master/logr.go#L294-L302) at debug level

Keep in mind, we wouldn't lose access to the underlying `logr.Logger`.

> Allowing you to e.g. inject a logr.Logger instance at a predefined level

To be clear, my concern so much about obtaining a logger that logs at a certain level of verbosity. My concern is the logr interface makes it harder to express the relative importance of an individual message.

Reply:
And lol at that link being almost exactly what I described.

Reply:
I don't have a philosophical attachment to the principles of `go-logr`. A lot of the stuff in the go-logr [FAQ](https://github.com/go-logr/logr?tab=readme-ov-file#why-v-levels) is quite principled. I'm not a go-logr purist so we don't have to fully accept their interface as-is. The main use case I was concerned with was that we didn't have a way of setting a logger to `sigs.k8s.io/controller-runtime/pkg/log`, which wants a `logr.Logger`.

I'm fine with creating the wrapper that embeds logr.Logger, especially if it maintains our code readability while also addressing the initial use case. The underlying `logr.Logger` is always available if we wish to use the lower interface.

FWIW `logger.Debug(...)` is definitely preferrable over `logger.V(logging.LevelDebug).Info(...)` and so if we had to switch all our logging statements to the 2nd version, that might be reason enough to create our wrapper.

Reply:
Thanks @jessesuen.

@bakunowski are you game for adding a wrapper like the one @hiddeco [linked to](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/b918c3636f5dcc030bbbc5191e757e37df87c682/pkg/logger/logger.go) in our existing `logging` package?